src/main/java/org/jfrog/hudson/ArtifactoryBuilder.java

```diff
@@ -381,16 +382,32 @@ private boolean isEmptyUrls(List<JFrogPlatformInstance> jfrogInstances) {
             return false;
         }
 
-        private boolean fillEmptyServers(List<JFrogPlatformInstance> jfrogInstances) {
+        private void autoFillPlatformServers(List<JFrogPlatformInstance> jfrogInstances) {
             if (jfrogInstances == null) {
-                return false;
+                return;
             }
             for (JFrogPlatformInstance instance : jfrogInstances) {
                 if (StringUtils.isBlank(instance.getArtifactoryServer().getArtifactoryUrl())) {
                     instance.getArtifactoryServer().setArtifactoryUrl(instance.getUrl() + "/artifactory");
+                    continue;
+                }
+                if (StringUtils.isBlank(instance.getUrl())) {
+                    continue;
+                }
+                // Check if Artifactory URL has a different prefix than platform URL.
+                if (!StringUtils.startsWithIgnoreCase(instance.getArtifactoryServer().getArtifactoryUrl(), instance.getUrl())) {
+                    // Try to search JFrog instance by the current Artifactory URL.
+                    Optional<JFrogPlatformInstance> oldInstance = this.jfrogInstances.stream().filter(currentInstance -> currentInstance.getArtifactoryServer().getArtifactoryUrl().equals(instance.getArtifactoryServer().getArtifactoryUrl()) && currentInstance.getId().equals(instance.getId())).findFirst();
+                    if (oldInstance.isPresent()) {
+                        // At this point, the Platform URL is different than the Artifactory URL.
+                        // Artifactory URL has a different prefix than Platform URL.
+                        // Artifactory URL has not changed, compared to last time configuration.
+                        // Since Artifactory URL is hidden in Jenkins UI under the advanced tab, the user may forget to change the Artifactory URL along with the platform.
+                        // As a result, override Artifactory url
```

Comment:
This comment may need some rephrasing, to make it clearer. I need to better understand the code flow first, before I can suggest the changes here.